engine/src/org/pentaho/di/job/entries/unzip/JobEntryUnZip.java

```diff
@@ -789,45 +789,7 @@ public boolean includeFile( FileSelectInfo info ) {
       // System.gc();
 
       // Unzip done...
-      if ( afterunzip == 1 ) {
-        // delete zip file
-        boolean deleted = fileObject.delete();
-        if ( !deleted ) {
-          updateErrors();
-          logError( BaseMessages.getString( PKG, "JobUnZip.Cant_Delete_File.Label", sourceFileObject.toString() ) );
-        }
-        // File deleted
-        if ( log.isDebug() ) {
-          logDebug( BaseMessages.getString( PKG, "JobUnZip.File_Deleted.Label", sourceFileObject.toString() ) );
-        }
-      } else if ( afterunzip == 2 ) {
-        FileObject destFile = null;
-        // Move File
-        try {
-          String destinationFilename = movetodir + Const.FILE_SEPARATOR + fileObject.getName().getBaseName();
-          destFile = KettleVFS.getFileObject( destinationFilename, this );
-
-          fileObject.moveTo( destFile );
-
-          // File moved
-          if ( log.isDetailed() ) {
-            logDetailed( BaseMessages.getString(
-              PKG, "JobUnZip.Log.FileMovedTo", sourceFileObject.toString(), realMovetodirectory ) );
-          }
-        } catch ( Exception e ) {
-          updateErrors();
-          logError( BaseMessages.getString(
-            PKG, "JobUnZip.Cant_Move_File.Label", sourceFileObject.toString(), realMovetodirectory, e
-              .getMessage() ) );
-        } finally {
-          if ( destFile != null ) {
-            try {
-              destFile.close();
-            } catch ( IOException ex ) { /* Ignore */
-            }
-          }
-        }
-      }
+      doUnzipPostprocessing( sourceFileObject, movetodir, realMovetodirectory );
```

Comment:
Would prefer the if ( afterunzip == 1 ) { to be around the call to doUnzipPostprocessing so that an unnecessary call is avoided.